rumqttc/src/eventloop.rs

```diff
@@ -156,7 +158,7 @@ impl EventLoop {
             o = network.readb(&mut self.state) => {
                 o?;
                 // flush all the acks and return first incoming packet
-                network.flush(&mut self.state.write).await?;
+                time::timeout(network_timeout, network.flush(&mut self.state.write)).await??;
```

Comment:
See if we can have more descriptive errors for flush timeouts

Reply:
i have added a variant for `network timeout` and other for `flush timeout`, https://github.com/bytebeamio/rumqtt/pull/545/files#diff-2d1780a6ddafae5dabfb4d74dea0621a46d8505f493a46981589205fffb4faedR34-R37. 

Do you think there can be any other kind of timeout's that we might need to add in future? 

If so then i think having a `Timeout` variant that contains an enum of different kind of variants (i.e. `Timeout(TimeoutType)`) which we can add to in future would be a better idea.